androidHyperskillApp/src/main/java/org/hyperskill/app/android/study_plan/fragment/StudyPlanFragment.kt

```diff
@@ -44,8 +46,42 @@ class StudyPlanFragment :
     private var gamificationToolbarDelegate: GamificationToolbarDelegate? = null
     private var studyPlanWidgetDelegate: StudyPlanWidgetDelegate? = null
 
+    private val onForegroundObserver =
+        object : DefaultLifecycleObserver {
+            override fun onPause(owner: LifecycleOwner) {
+                super.onPause(owner)
+                wasInBackground = true
+            }
+        }
+
+    private var wasInPause = false
+    private var wasInBackground = false
+
+    override fun onPause() {
+        super.onPause()
+        wasInPause = true
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (wasInBackground) {
+            studyPlanViewModel.onNewMessage(
+                StudyPlanScreenFeature.Message.Initialize(forceUpdate = true)
+            )
+        } else if (wasInPause) {
+            studyPlanViewModel.onNewMessage(
+                StudyPlanScreenFeature.Message.StudyPlanWidgetMessage(
+                    StudyPlanWidgetFeature.Message.ReloadContentInBackground
+                )
+            )
+        }
+        wasInPause = false
+        wasInBackground = false
+    }
```

Comment:
But for track coming to the background yu should use `onStop` lifecycle event. See Fragment lifecycle guid https://developer.android.com/guide/fragments/lifecycle and Activity lifecycle guid https://developer.android.com/guide/components/activities/activity-lifecycle.

`wasInPause` and `wasInBackground` are triggered by the same lifecycle event - `onPause`, so `StudyPlanWidgetFeature.Message.ReloadContentInBackground` will never be sent.

Generally, it's better to create a special LifecycleObserver class, to be notified about return from background or other custom events and hide local mutable flags.